src/plugins/telemetry/public/plugin.ts

```diff
@@ -146,7 +156,14 @@ export class TelemetryPlugin implements Plugin<TelemetryPluginSetup, TelemetryPl
 
     analytics.registerContextProvider({
       name: 'telemetry labels',
-      context$: of({ labels: this.config.labels }),
+      context$: this.telemetryLabels$.pipe(
+        tap((labels) => {
+          // Hack to update the APM agent's labels.
+          // In the future we might want to expose APM as a core service to make reporting metrics much easier.
+          window.elasticApm?.addLabels(labels);
```

Comment:
For the UI, the agent is global, so `addLabels` works great.

However, on the server, the nodejs agent needs the global labels setter because `addLabels` only applies to the current transaction.